# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/xfdesktop/default-t2-background.patch
# Copyright (C) 2024 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

--- xfdesktop-4.20.0/src/xfdesktop-backdrop-manager.c.vanilla	2024-12-20 21:44:15.123949922 +0100
+++ xfdesktop-4.20.0/src/xfdesktop-backdrop-manager.c	2024-12-20 22:07:59.577140810 +0100
@@ -472,22 +472,19 @@
     }
 }
 
-static GdkRGBA
-fetch_color(XfconfChannel *channel, const gchar *property_name) {
-    GdkRGBA color;
+static void
+fetch_color(XfconfChannel *channel, const gchar *property_name, GdkRGBA* color) {
 
     if (!xfconf_channel_get_array(channel,
                                   property_name,
-                                  G_TYPE_DOUBLE, &color.red,
-                                  G_TYPE_DOUBLE, &color.green,
-                                  G_TYPE_DOUBLE, &color.blue,
-                                  G_TYPE_DOUBLE, &color.alpha,
+                                  G_TYPE_DOUBLE, &color->red,
+                                  G_TYPE_DOUBLE, &color->green,
+                                  G_TYPE_DOUBLE, &color->blue,
+                                  G_TYPE_DOUBLE, &color->alpha,
                                   G_TYPE_INVALID))
     {
-        color.red = color.green = color.blue = color.alpha = 1.0;
+        //color.red = color.green = color.blue = color.alpha = 1.0;
     }
-
-    return color;
 }
 
 static XfwWorkspace *
@@ -882,15 +879,17 @@
     DBG("Creating backdrop from setting prefix %s", property_prefix);
 
     prop_name = g_strconcat(property_prefix, "/color-style", NULL);
-    XfceBackdropColorStyle color_style = xfconf_channel_get_int(channel, prop_name, XFCE_BACKDROP_COLOR_TRANSPARENT);
+    XfceBackdropColorStyle color_style = xfconf_channel_get_int(channel, prop_name, XFCE_BACKDROP_COLOR_VERT_GRADIENT);
     g_free(prop_name);
 
     prop_name = g_strconcat(property_prefix, "/rgba1", NULL);
-    GdkRGBA color1 = fetch_color(channel, prop_name);
+    GdkRGBA color1 = {0.122, 0.0666, 0.216, 1.0};
+    fetch_color(channel, prop_name, &color1);
     g_free(prop_name);
 
     prop_name = g_strconcat(property_prefix, "/rgba2", NULL);
-    GdkRGBA color2 = fetch_color(channel, prop_name);
+    GdkRGBA color2 = {0.0392, 0.0235, 0.0705, 1.0};
+    fetch_color(channel, prop_name, &color2);
     g_free(prop_name);
 
     prop_name = g_strconcat(property_prefix, "/image-style", NULL);
